libminifi/include/core/ProcessSession.h

```diff
@@ -109,8 +109,8 @@ class ProcessSession : public ReferenceContainer {
    */
   void importFrom(io::DataStream &stream, const std::shared_ptr<core::FlowFile> &flow);
   // import from the data source.
-  void import(std::string source, const std::shared_ptr<core::FlowFile> &flow, bool keepSource = true, uint64_t offset = 0);
-  void import(std::string source, std::vector<std::shared_ptr<FlowFileRecord>> &flows, bool keepSource, uint64_t offset, char inputDelimiter);
```

Comment:
You're changing the contract. We will need to keep the (...bool,uint64_t,char) variant.

Reply:
adding variants is not an issue, though.

Reply:
You are right, as this change would break API, and should not be made.
However I would prefer to eventually move to this variant, as removing the file in this case does not make much sense (we are reading "lines" and if there is a partial line at the end of the file, it is not ingested, and removing the file would cause data loss). By the way, we do not use this any other way than keeping the file.